From 6e0848a1c51c6494e3b7410c5fe38941d48fcb36 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois-Simon=20Fauteux-Chapleau?=
 <francois-simon.fauteux-chapleau@savoirfairelinux.com>
Date: Wed, 16 Oct 2024 22:32:12 -0400
Subject: [PATCH] qtwebengine: fix v8 build error

In file included from ../../../3rdparty/chromium/v8/src/heap/cppgc/sweeper.h:14,
                 from ./../../../3rdparty/chromium/v8/src/heap/cppgc/sweeper.cc:5,
                 from gen/v8/cppgc_base_jumbo_7.cc:5:
../../../3rdparty/chromium/v8/src/heap/cppgc/stats-collector.h: In member function 'void cppgc::internal::StatsCollector::ForAllAllocationObservers(Callback)':
../../../3rdparty/chromium/v8/src/heap/cppgc/stats-collector.h:401:48: error: cannot convert 'std::vector&lt;cppgc::internal::StatsCollector::AllocationObserver*&gt;::iterator' to 'const char*'
  401 |         std::remove(allocation_observers_.begin(), allocation_observers_.end(),
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
      |                                                |
      |                                                std::vector&lt;cppgc::internal::StatsCollector::AllocationObserver*&gt;::iterator
---
 .../src/3rdparty/chromium/v8/src/heap/cppgc/stats-collector.h    | 1 +
 1 file changed, 1 insertion(+)

diff --git a/qtwebengine/src/3rdparty/chromium/v8/src/heap/cppgc/stats-collector.h b/qtwebengine/src/3rdparty/chromium/v8/src/heap/cppgc/stats-collector.h
index 2cf728489d..d8414ae3c6 100644
--- a/qtwebengine/src/3rdparty/chromium/v8/src/heap/cppgc/stats-collector.h
+++ b/qtwebengine/src/3rdparty/chromium/v8/src/heap/cppgc/stats-collector.h
@@ -8,6 +8,7 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#include <algorithm>
 #include <atomic>
 #include <vector>
 
-- 
2.47.0

